### ISSUE TITLE
graph_from_string

### ISSUE THREAD

**Issue by ggrothendieck** [igraph/rigraph#475]
graph_from_literal requires an unevaluated argument which is difficult to deal with.  graph_from_string which would take a character string as input would be helpful.  See https://stackoverflow.com/questions/69252290/pass-a-string-as-literal-to-create-a-graph-object/69252447#69252447 for an implementation.

**Comment by szhorvat** [comment]
The Mathematica and Python interfaces both use string inputs, and have a fair degree of cross-compatibility. Thus it might make sense to provide the same for R.

**Comment by szhorvat** [comment]
@ggrothendieck Can you please show an example that demonstrates the difficulty with the current function, and how undesirable evaluation happens? This would be useful for those of us less educated in R.

**Comment by szhorvat** [comment]
I'll copy here the original motivation for wanting to use strings from the just-updated StackOverflow question:

```
# input
c('0.1', '0.1.1', '0.1.2', '0.11', '0.12', '0.11.1', '0.12.1', '0.12.2')

# expected output:
graph_from_literal(0-1, 0-1-1, 0-1-2, 0-11, 0-12, 0-11-1, 0-12-1, 0-12-2)
```

I.e. [this problem](https://stackoverflow.com/q/64440386/695132) might be solved by replacing `.`s with `-`s and joining strings.

**Comment by zx8754** [comment]
@ggrothendieck Thanks for posting the issue here. I will keep an eye on updates.

**Comment by gaborcsardi** [comment]
Do not use `graph_from_literal` to create graphs programmatically, it is not meant for that. Put your edge list + attributes in a data frame, and create a graph from that.

**Comment by zx8754** [comment]
@gaborcsardi agreed, "literal" is not meant for this task. But given above input, how would you suggest to convert it into a graph object?

I understand that this kind of input might not be popular graph input, and there isn't a real need for dedicated `graph_from_string` function.

**Comment by gaborcsardi** [comment]
> @gaborcsardi agreed, "literal" is not meant for this task. But given above input, how would you suggest to convert it into a graph object?

Split them with `strsplit()`, then create a data frame from each, and then `rbind()` them together, then use `graph_from_data_frame()`.

**Comment by szhorvat** [comment]
I will close this issue now, since there seems to be a consensus that the proposed use of `graph_from_literal` would not be appropriate (I agree about this as well), and no other use case has been put forward for string inputs.
